### PR TITLE
JDK-8296968: Update langtools tests to use @enablePreview

### PR DESCRIPTION
**Pull request by jddarcy**
Update langtools tests to use JTreg's @enablePreview tag, where appropriate.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296968](https://bugs.openjdk.org/browse/JDK-8296968): Update langtools tests to use @enablePreview

### Reviewers
 * [Athijegannathan Sundararajan](https://openjdk.org/census#sundar) (@sundararajana - **Reviewer**) ⚠️ Review applies to [9fa41e15](https://git.openjdk.org/jdk/pull/11149/files/9fa41e15a86efcb6c7fd7f5faa6104337d91356c)
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11149/head:pull/11149` \
`$ git checkout pull/11149`

Update a local copy of the PR: \
`$ git checkout pull/11149` \
`$ git pull https://git.openjdk.org/jdk pull/11149/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11149`

View PR using the GUI difftool: \
`$ git pr show -t 11149`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11149.diff">https://git.openjdk.org/jdk/pull/11149.diff</a>

</details>

### CODE REVIEW

**Review by sundararajana — APPROVED** [review]
LGTM

**Review by JimLaskey — COMMENTED** [review]
Just as a note: when using @enablePreview it is unnecessary to use @run main/othervm ... . That is the @run statement can be dropped.

File: test/langtools/tools/javac/annotations/typeAnnotations/classfile/Patterns.java [hunk 1]
```diff
@@ -28,8 +28,8 @@
  * @modules java.compiler
  *          jdk.jdeps/com.sun.tools.javap
  * @build toolbox.JavapTask
- * @compile --enable-preview -source ${jdk.version} Patterns.java
- * @run main/othervm --enable-preview Patterns
+ * @enablePreview
+ * @run main/othervm Patterns
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/annotations/typeAnnotations/classfile/Patterns.java:32]: do we still need `/othervm` ?

File: test/langtools/tools/javac/patterns/EnumTypeChanges.java [hunk 2]
```diff
@@ -25,9 +25,10 @@
  * @test
  * @bug 8262891
  * @summary Verify pattern switches work properly when the set of enum constant changes.
- * @compile --enable-preview -source ${jdk.version} EnumTypeChanges.java
- * @compile --enable-preview -source ${jdk.version} EnumTypeChanges2.java
- * @run main/othervm --enable-preview EnumTypeChanges
+ * @enablePreview
+ * @compile EnumTypeChanges.java
+ * @compile EnumTypeChanges2.java
+ * @run main/othervm EnumTypeChanges
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/EnumTypeChanges.java:31]: Do we need `/othervm` ?

File: test/langtools/tools/javac/patterns/PatternErrorRecovery.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
  * @test /nodynamiccopyright/
  * @bug 8268320
  * @summary Verify user-friendly errors are reported for ill-formed pattern.
-* @compile/fail/ref=PatternErrorRecovery.out -XDrawDiagnostics -XDshould-stop.at=FLOW --enable-preview -source ${jdk.version} PatternErrorRecovery.java
+ * @compile/fail/ref=PatternErrorRecovery.out -XDrawDiagnostics -XDshould-stop.at=FLOW --enable-preview -source ${jdk.version} PatternErrorRecovery.java
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/PatternErrorRecovery.java:5]: why no `@enablePreview` for this test?

File: test/langtools/tools/javac/patterns/SealedTypeChanges.java [hunk 4]
```diff
@@ -25,15 +25,15 @@
  * @test
  * @bug 8262891 8270151
  * @summary Verify pattern switches work properly when the set of sealed types changes.
- * @compile --enable-preview -source ${jdk.version} SealedTypeChanges.java
- * @compile --enable-preview -source ${jdk.version} SealedTypeChanges2.java
- * @run main/othervm --enable-preview SealedTypeChanges
+ * @enablePreview
+ * @compile SealedTypeChanges.java
+ * @compile SealedTypeChanges2.java
+ * @run main/othervm SealedTypeChanges
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/SealedTypeChanges.java:31]: `/othervm` ?

File: test/langtools/tools/javac/patterns/SimpleAndGuardPattern.java [hunk 5]
```diff
@@ -23,8 +23,9 @@
 
 /**
  * @test
- * @compile --enable-preview -source ${jdk.version} -doe SimpleAndGuardPattern.java
- * @run main/othervm --enable-preview SimpleAndGuardPattern
+ * @enablePreview
+ * @compile -doe SimpleAndGuardPattern.java
+ * @run main/othervm SimpleAndGuardPattern
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/SimpleAndGuardPattern.java:28]: `/othervm` ?

File: test/langtools/tools/javac/patterns/SwitchErrors.out [hunk 6]
```diff
@@ -1,58 +1,58 @@
...
+SwitchErrors.java:82:9: compiler.err.not.exhaustive.statement
+SwitchErrors.java:87:9: compiler.err.not.exhaustive.statement
+SwitchErrors.java:92:9: compiler.err.not.exhaustive.statement
+SwitchErrors.java:98:9: compiler.err.not.exhaustive.statement
+SwitchErrors.java:105:9: compiler.err.not.exhaustive.statement
+SwitchErrors.java:159:9: compiler.err.not.exhaustive.statement
+SwitchErrors.java:232:9: compiler.err.not.exhaustive.statement
 - compiler.note.preview.filename: SwitchErrors.java, DEFAULT
 - compiler.note.preview.recompile
-55 errors
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/patterns/SwitchErrors.out:58]: Interesting glyph for no trailing newline; I've not seen that before

File: test/langtools/tools/javac/patterns/NestedPrimitiveDeconstructionPattern.java [hunk 7]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (lahodaj) [inline test/langtools/tools/javac/patterns/NestedPrimitiveDeconstructionPattern.java:1]: Nit - maybe add the empty line behind or before the `@test` comment, just to be sure header checkers are not confused?

**Review by jonathan-gibbons — COMMENTED** [review]
In various places `/othervm` has been left in place, but not clear why.

In one place, whitespace was fixed, but `@enablePreview` was not added

File: test/langtools/tools/javac/annotations/typeAnnotations/classfile/Patterns.java (hunk 1 above)
Comment (jddarcy) [reply test/langtools/tools/javac/annotations/typeAnnotations/classfile/Patterns.java:32]: I suppose it isn't strictly required; I assume the @enablePreview functionality handles this if needed. I'll go through and remove the /othervm options and make sure the tests still pass, etc.

File: test/langtools/tools/javac/patterns/PatternErrorRecovery.java (hunk 3 above)
Comment (jddarcy) [reply test/langtools/tools/javac/patterns/PatternErrorRecovery.java:5]: The test has multiple @compile directives, one of which does not use --enable-preview. (I didn't check closely to see if enablePreview could be toggled on/off for different sets of directives.)

File: test/langtools/tools/javac/patterns/NestedPrimitiveDeconstructionPattern.java (hunk 7 above)
Comment (jddarcy) [reply test/langtools/tools/javac/patterns/NestedPrimitiveDeconstructionPattern.java:1]: Okay -- must have accidentally added a blank line.

File: test/langtools/tools/javac/patterns/PatternErrorRecovery.java (hunk 3 above)
Comment (jonathan-gibbons) [reply test/langtools/tools/javac/patterns/PatternErrorRecovery.java:5]: You can't toggle. Thanks for the note.

**Review by jonathan-gibbons — APPROVED** [review]
Nice cleanup

**Pull request closed without merging**